Orckestra.Widget.FilteredSelector/Orckestra.Widget.FilteredSelector/WidgetProvider/FilteredSelectorWidgetProvider.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Composite.Data;
+using Composite.Functions;
+using Composite.Functions.Plugins.WidgetFunctionProvider;
+using Composite.Plugins.Functions.WidgetFunctionProviders.StandardWidgetFunctionProvider.Foundation;
+
+using Microsoft.Practices.EnterpriseLibrary.Common.Configuration;
+using Microsoft.Practices.EnterpriseLibrary.Common.Configuration.ObjectBuilder;
+using Microsoft.Practices.ObjectBuilder;
+
+using Orckestra.Widget.FilteredSelector.WidgetProvider.Functions;
+
+namespace Orckestra.Widget.FilteredSelector.WidgetProvider
+{
+    [ConfigurationElementType(typeof(FilteredSelectorWidgetFunctionProviderData))]
+    internal sealed class FilteredSelectorWidgetProvider : IDynamicTypeWidgetFunctionProvider
+    {
+        public FilteredSelectorWidgetProvider(string providerName) => _entityTokenFactory = new EntityTokenFactory(providerName);
+        private readonly EntityTokenFactory _entityTokenFactory = null;
+        private List<IWidgetFunction> _widgetDynamicTypeFunctions = null;
+        public IEnumerable<IWidgetFunction> Functions { get; }
+        public WidgetFunctionNotifier WidgetFunctionNotifier { get; set; }
+        private void InitializeDynamicTypeFunctions()
+        {
+            _widgetDynamicTypeFunctions = new List<IWidgetFunction>();
+
+            IEnumerable<Type> dataInterfaces = DataFacade.GetAllKnownInterfaces(UserType.Developer)
+                .Where(x => typeof(IPageRelatedData).IsAssignableFrom(x));

```

Comment:
you can even go 
`.Where(typeof(IPageRelatedData).IsAssignableFrom)`

Reply:
Thanks, updated